File: src/main/java/com/github/nsc/de/compiler/parser/node/functions/FunctionArgumentNode.java
package com.github.nsc.de.compiler.parser.node.functions;

import com.github.nsc.de.compiler.parser.node.Node;

public class FunctionArgumentNode implements Node {

    private final String name;

    public FunctionArgumentNode(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "FunctionArgumentNode{" +
                "name='" + name + '\'' +
                '}';
    }
}

File: shake/compiler/parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/functions/FunctionArgumentNode.kt
package io.github.shakelang.shake.parser.node.functions

import io.github.shakelang.shake.parser.node.VariableType
import io.github.shakelang.shake.parser.node.Node
import io.github.shakelang.parseutils.characters.position.PositionMap
import kotlin.jvm.JvmOverloads

class FunctionArgumentNode @JvmOverloads constructor(
    map: PositionMap,
    val name: String,
    private val type: VariableType = VariableType.DYNAMIC
) : Node(map) {

    override fun toJson(): Map<String, *> =
        mapOf("name" to "FunctionArgumentNode", "argument_name" to name, "type" to type.toString())

}